src/lib/workspace/AnalysisPanel.tsx

```diff
@@ -175,10 +177,13 @@ export function AnalysisPanel(props: Props) {
           <Subsetting {...props.match.params} analysisState={analysisState} />
         )}
       />
-      <Route
-        path={`${routeBase}/visualizations`}
-        render={() => <ComputationRoute analysisState={analysisState} />}
-      />
+      {/* use ShowHideVariableContextProvider here for handling the state of show/hide variable toggle */}
+      <ShowHideVariableContextProvider>
+        <Route
+          path={`${routeBase}/visualizations`}
+          render={() => <ComputationRoute analysisState={analysisState} />}
+        />
+      </ShowHideVariableContextProvider>
```

Comment:
The provider should be the root node returned by this component. Otherwise the state will be reset every time the user switches between the subset/viz tabs.

Reply:
@dmfalke actually I wrapped whole returning div with this provider at first and then changed it to a local region (viz only). Will change this back 😄